tests/phpunit/tests/rest-api/wpRestBlockPatternsController.php

```diff
@@ -0,0 +1,171 @@
+
+	/**
+	 * Set up class test fixtures.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @param WP_UnitTest_Factory $factory WordPress unit test factory.
+	 */
+	public static function wpSetUpBeforeClass( $factory ) {
+		self::$admin_id = $factory->user->create( array( 'role' => 'administrator' ) );
+
+		// Setup an empty testing instance of `WP_Block_Patterns_Registry` and save the original.
+		self::$orig_registry              = WP_Block_Patterns_Registry::get_instance();
+		self::$registry_instance_property = new ReflectionProperty( 'WP_Block_Patterns_Registry', 'instance' );
+		self::$registry_instance_property->setAccessible( true );
+		$test_registry = new WP_Block_Pattern_Categories_Registry();
+		self::$registry_instance_property->setValue( $test_registry );
+
+		// Register some patterns in the test registry.
+		$test_registry->register(
+			'test/one',
+			array(
+				'title'         => 'Pattern One',
+				'categories'    => array( 'test' ),
+				'viewportWidth' => 1440,
+				'content'       => '<!-- wp:heading {"level":1} --><h1>One</h1><!-- /wp:heading -->',
+			)
+		);
+
+		$test_registry->register(
+			'test/two',
+			array(
+				'title'      => 'Pattern Two',
+				'categories' => array( 'test' ),
+				'content'    => '<!-- wp:paragraph --><p>Two</p><!-- /wp:paragraph -->',
+			)
+		);
+	}
+
+	public static function wpTearDownAfterClass() {
+		self::delete_user( self::$admin_id );
+
+		// Restore the original registry instance.
+		self::$registry_instance_property->setValue( self::$orig_registry );
+		self::$registry_instance_property->setAccessible( false );
+		self::$registry_instance_property = null;
+		self::$orig_registry              = null;
+	}
+
+	public function set_up() {
+		parent::set_up();
+
+		switch_theme( 'emptytheme' );
+	}
+
+	public function test_register_routes() {
+		$routes = rest_get_server()->get_routes();
+		$this->assertArrayHasKey( static::REQUEST_ROUTE, $routes );
+	}
+
+	public function test_get_items() {
+		wp_set_current_user( self::$admin_id );
+
+		$request            = new WP_REST_Request( 'GET', static::REQUEST_ROUTE );
+		$request['_fields'] = 'name,content';
+		$response           = rest_get_server()->dispatch( $request );
+		$data               = $response->get_data();
+
+		$this->assertIsArray( $data, 'WP_REST_Block_Patterns_Controller::get_items() should return an array' );
+		$this->assertGreaterThanOrEqual( 2, count( $data ), 'WP_REST_Block_Patterns_Controller::get_items() should return at least 2 items' );
+		$this->assertSame(
+			array(
+				'name'    => 'test/one',
+				'content' => '<!-- wp:heading {"level":1} --><h1>One</h1><!-- /wp:heading -->',
+			),
+			$data[0],
+			'WP_REST_Block_Patterns_Controller::get_items() should return test/one'
+		);
+		$this->assertSame(
+			array(
+				'name'    => 'test/two',
+				'content' => '<!-- wp:paragraph --><p>Two</p><!-- /wp:paragraph -->',
+			),
+			$data[1],
+			'WP_REST_Block_Patterns_Controller::get_items() should return test/two'
+		);
+	}
```

Comment:
Same as the previous controller about a permission test.

Reply:
✅ See 0b89e425e7.